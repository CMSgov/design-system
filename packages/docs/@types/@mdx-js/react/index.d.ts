declare module '@mdx-js/react' {
  const MDXProvider: React.ComponentType;
  export { MDXProvider };
}
