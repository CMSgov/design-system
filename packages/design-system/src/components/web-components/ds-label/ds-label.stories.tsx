import WebComponentDocTemplate from '../../../../../../.storybook/docs/WebComponentDocTemplate.mdx';
import { webComponentDecorator } from '../storybook';
import './ds-label';

export default {
  title: 'Web Components/ds-label',
  argTypes: {
    'text content': {
      control: 'text',
      controlsOnly: true,
    },
    inversed: {
      description: 'Set to "true" to apply the "inverse" theme',
      control: 'boolean',
    },
    for: {
      control: 'text',
      description: 'The ID of the field this label is for',
    },
    'root-id': {
      control: 'text',
      description:
        "A unique ID for this element. A unique ID will be generated if one isn't provided.",
    },
  },
  args: {
    'text content': 'A date of birth cannot be in the future.',
  },
  parameters: {
    docs: {
      page: WebComponentDocTemplate,
      description: {
        component: `For information about how and when to use this component, [refer to its full documentation page](https://design.cms.gov/components/label/).`,
      },
    },
  },
  decorators: [webComponentDecorator],
};

const Template = ({ 'text content': text, ...args }) => <ds-label {...args}>{text}</ds-label>;

export const Default = Template.bind({});
