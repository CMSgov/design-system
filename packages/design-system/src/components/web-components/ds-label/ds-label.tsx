import { define } from '../preactement/define';
import { Label, LabelProps } from '../../Label';
import { parseBooleanAttr } from '../wrapperUtils';

const attributes = ['class-name', 'inversed', 'for', 'root-id'];

interface WrapperProps extends Omit<LabelProps, 'inversed' | 'fieldId'> {
  inversed?: string;
  rootId?: string;
  for?: string;
}

const Wrapper = ({ inversed, rootId, for: fieldId, ...otherProps }: WrapperProps) => (
  <Label {...otherProps} inversed={parseBooleanAttr(inversed)} fieldId={fieldId} id={rootId} />
);

define('ds-label', () => Wrapper, { attributes });
