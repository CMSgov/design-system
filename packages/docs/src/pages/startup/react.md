---
title: Using React components
weight: 3
---

The examples below assume you've installed the design system using NPM and have already setup your build system.

<h4 class="ds-h4 ds-u-font-size--base">Default imports</h4>

Individual components can be imported from their individual export file.

```jsx
import Button from '@cmsgov/design-system-core/dist/components/Button/Button';
```

<h4 class="ds-h4 ds-u-font-size--base">Named imports</h4>

Components can also be imported using the shorter syntax below.

<div class="ds-c-alert ds-c-alert--warn ds-u-margin-bottom--2 ds-u-font-size--small">
  <div class="ds-c-alert__body">
    <h3 class="ds-c-alert__heading ds-u-font-size--base">Performance note</h3>
    <p class="ds-c-alert__text">
      This approach may result in a much larger file than you intend. Depending on what module bundler you use, all of the design system's React components may be included in the bundled file even if you didn't specifically import them. This can be avoided by enabling features like [tree shaking in Webpack](https://webpack.js.org/guides/tree-shaking/).
    </p>
  </div>
</div>

```jsx
import { Button } from '@cmsgov/design-system-core';
```

[View an example](https://github.com/CMSgov/design-system/blob/master/examples/react-app/src/scripts/index.js)

<h2 id="examples" class="ds-h2 ds-u-color--primary-darker">Examples</h2>

Additional examples of the design system in use can be viewed on GitHub. These projects demonstrate the various ways you can incorporate the design system into your development process and various use cases.

<a href="https://github.com/CMSgov/design-system/tree/master/examples/" class="ds-c-button">Browse example projects</a>

<h2 id="need-help" class="ds-h2 ds-u-color--primary-darker">Need help or ran into an issue?</h2>

If you're having trouble installing or setting up the design system, or if you think you've found a bug, feel free to [open an issue on GitHub](https://github.com/CMSgov/design-system/issues).
