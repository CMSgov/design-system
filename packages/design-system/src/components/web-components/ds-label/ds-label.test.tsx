<<<<<<< HEAD
import { render, screen } from '@testing-library/react';
=======
import { createTestRenderer } from '../__tests__/rendering';
>>>>>>> 107c1cedb (Remove unused dep)
import './ds-label';

/* eslint-disable @typescript-eslint/no-namespace */
declare global {
  namespace JSX {
    interface IntrinsicElements {
      'ds-label': any;
    }
  }
}
/* eslint-enable */

const defaultProps = {
  for: 'field-123',
};

function renderLabel(props: any = defaultProps) {
  return render(<ds-label {...props}>Foo</ds-label>);
}

describe('Label', () => {
  it('should render a basic label', () => {
    const { asFragment } = renderLabel();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should apply an inverse class', () => {
    renderLabel({ inversed: true });
    const label = screen.getByText('Foo');
    expect(label.className).toContain('ds-c-label--inverse');
  });

  it('should apply custom classes', () => {
    renderLabel({ 'class-name': 'bar' });
    const label = screen.getByText('Foo');
    expect(label.className).toContain('bar');
  });
});
