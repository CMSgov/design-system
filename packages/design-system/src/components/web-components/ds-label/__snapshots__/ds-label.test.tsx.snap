// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render a basic label 1`] = `
<DocumentFragment>
  <ds-label
    for="field-123"
  >
    <label
      class="ds-c-label"
      for="field-123"
      parent="[object HTMLElement]"
    >
      Foo
    </label>
  </ds-label>
</DocumentFragment>
`;
